pyasic/miners/hammer/blackminer/DX/D10.py

```diff
@@ -3,4 +3,5 @@
 
 
 class HammerD10(BlackMiner, D10):
+    expected_hashrate = 5000
```

Comment:
This attribute doesn't exist?  I usually expect to query this from the API, as it can change with tuning and such.

Reply:
sadly they don't seem to have it populated, seems to be a subset


```
[D10 (Stock): 10.5.0.124]

{'STATUS': [{'STATUS': 'S', 'When': 1735059617, 'Code': 11, 'Msg': 'Summary', 'Description': 'ccminer 2.3.3'}], 'SUMMARY': [{'Elapsed': 763862, 'MHS 5s': '4793.9428', 'MHS av': '4843.4119', 'Found Blocks': 0, 'Getworks': 29647, 'Accepted': ' 68.0000', 'Rejected': ' 68.0000', 'Hardware Errors': 78910, 'Utility': 2.18, 'Discarded': 423803, 'Stale': 0, 'Get Failures': 0, 'Local Work': 26142192, 'Remote Failures': 0, 'Network Blocks': 5138, 'Total MH': 3699700115.8494, 'Work Utility': 4467223.75, 'Difficulty Accepted': 56735039488.0, 'Difficulty Rejected': 137363456.0, 'Difficulty Stale': 0.0, 'Best Share': 2593581, 'Device Hardware%': 0.0001, 'Device Rejected%': 0.2415, 'Pool Rejected%': 0.2415, 'Pool Stale%': 0.0, 'Last getwork': 1735059617}], 'id': 1}

{'STATUS': [{'STATUS': 'S', 'When': 1735059617, 'Code': 70, 'Msg': 'CGMiner stats', 'Description': 'ccminer 2.3.3'}], 'STATS': [{'CGMiner': '2.3.3', 'Miner': '2.0.2.2', 'CompileTime': '2023-05-04 15-34-32 CST', 'Type': 'Hammer D10'}, {'STATS': 0, 'ID': 'A30', 'Elapsed': 763862, 'Calls': 0, 'Wait': 0.0, 'Max': 0.0, 'Min': 99999999.0, 'MHS 5s': '4793.9428', 'MHS av': '4843.4119', 'miner_count': 3, 'frequency': '805', 'fan_num': 2, 'fan1': 5010, 'fan2': 4890, 'fan3': 0, 'fan4': 0, 'temp_num': 12, 'temp1': 33, 'temp2': 40, 'temp3': 34, 'temp4': 0, 'temp_max': 53, 'Device Hardware%': 0.0, 'no_matching_work': 78910, 'chain_acn1': 108, 'chain_acn2': 108, 'chain_acn3': 108, 'chain_acn4': 0, 'chain_acs1': ' oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooo', 'chain_acs2': ' oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooo', 'chain_acs3': ' oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooooooo oooooooooooooooo oooooooo oooooooo oooooooo oooooooo oooo', 'chain_acs4': '', 'chain_hw1': 31690, 'chain_hw2': 4504, 'chain_hw3': 42716, 'chain_hw4': 0, 'chain_rate1': '1691.0335', 'chain_rate2': '1546.0866', 'chain_rate3': '1556.8227', 'chain_rate4': ''}], 'id': 1}
```

Reply:
That's fine, maybe just remove it then?

Reply:
Added it so percent_expected_hashrate would get populated, but we can remove it if you prefer

Reply:
Maybe change it to `sticker_hashrate` and allow fallback to that if the `expected_hashrate` is None?

Reply:
seems reasonable, let me make that change

Reply:
pushed the concept of sticker_hashrate, since its really only applies to the D10 tried to make it as non-intrusive as possible. sticker_hashrate as you mentioned will be hard to track for most machine models as they sell models at multiple hashrates.